### ISSUE TITLE
Some remarks about the file 'about_Plaster_CreatingAManifest.help.md' 

### ISSUE THREAD

**Issue by LaurentDardenne** [PowerShell/Plaster#164]
**LONG DESCRIPTION**
 _"This topic serves to document the existing manifest schema, currently version 0.4, ..."_
 May be add a link to the .xsd file ?

**Metadata**
 Explain how works the versionning in Plaster: https://github.com/PowerShell/Plaster/issues/89

**Parameters**
 Explain the behavoir of the 'store' parameter.

Typo in "Parameter Type: Text" the last example is wrong :
  _Enter the version number for the module (**1.0.0**):_
 to
  _Enter the version number for the module (**0.1.0**):_

**Content :**
 _"file - Specify one or more files to copy under the destination folder."_
 
 Explain the constraint : we can not copy outside the destination folder
 
 note: we must use 'TemplateFile' to preserve the original encoding 
```Powershell
       <!-- We must use Encoding attribut : Powershell do not preserve the original encoding -->
    <templateFile source='DirectoriesProject\**'  destination='' encoding='UTF8'/>
```

_"templateFile - Specify one or more template files to copy and expand under the destination folder."_
 May be add a example here. We must use ' * ' or '**' ?

**Content element: XXX**
 Condition is empty (todo ?)   

_TODO: Talk about the Plaster parameters._
 Explain where $Plaster_xx variables comes from (dynamics parameters). 
 Explicit note : All variables can be used in a Plaster manifest.

**Content element: Modify**
 _"original - The original text, or regular expression match to replace."_
 Must be escaped the simple text : 'C:\Temp' or 'C:\\\Temp' ?

 Question : The  '(?s)' option must be allways specified ?

**Content element: RequireModule**
 May be add recommendation ?
 See [Best practice](https://github.com/PoshCode/PowerShellPracticeAndStyle/issues/68) 
 _7. ALWAYS use three digits for versions (i.e. 5.0.0, not 5.0)_

**Comment by davegreen** [comment]
Thanks @LaurentDardenne, I'll continue my docs branch to expand on this feedback and make this about file better.

> note: we must use 'TemplateFile' to preserve the original encoding

Interesting. I was under the impression that the existing encoding was preserved if no encoding was specified, but since Plaster is opening and saving the file, I can see how that would happen. Do you know what the default is if you do not specify an encoding? I imagine it would be UTF-8?

> Question : The '(?s)' option must be allways specified ?

I'm not sure. I don't think I'm entirely sure how that works, I didn't think it was valid regex, but it appears to match any prefix to your intended match. I'll do some testing to educate myself :).

**Comment by rkeithhill** [comment]
> note: we must use 'TemplateFile' to preserve the original encoding

`<file>` now preserves the original encoding because it never does anything to the file other than copy it.  `<templateFile>` was created specifically to address this problem i.e. the only files specified as a `<templateFile>` are the files that have substitution placeholders that need to be substituted.  For these files we allow the user to specify the encoding the want the written file to use.

The default encoding is `Default` which for the `Set-Content` cmdlet is ASCII.  I'm open to changing this but if we do UTF-8 some (Linux) folks will complain that it has a BOM and others would complain it doesn't.  :-)

**Comment by rkeithhill** [comment]
> Must be escaped the simple text : 'C:\Temp' or 'C:\\Temp' ?

Yes, you have to escape the backslash because the text represents a regex pattern that will be evaluated against the text of the file.  

> Question : The '(?s)' option must be allways specified ?

Only use `(?[smi])` if you need to override PowerShell's regex default behavior.  `(?s)` means that `.` matches every char *including* `\n`.  This is how you can write a regex to span multiple lines.

**Comment by davegreen** [comment]
Closed with PR #179.
